### PR TITLE
Fix moving mouse off playback bar while scrubbing

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
Not worth calling out in release notes

**Description**

- Allows scrubbing to continue even if the mouse moves outside the playback bar or window
- Tooltip remains visible the entire time

I made this work in https://github.com/foxglove/studio/pull/2912, then I broke it again in https://github.com/foxglove/studio/pull/4514. The pendulum swings back again.

Additionally I believe the behavior of showing the tooltip even when the mouse is outside the bar changed in https://github.com/foxglove/studio/pull/4392.

Before

https://github.com/foxglove/studio/assets/14237/620511de-f16d-446f-a538-11a80373e186

After

https://github.com/foxglove/studio/assets/14237/98b41e2a-4732-4e77-8362-87683d385625

### CODE REVIEW

File: packages/studio-base/src/components/PlaybackControls/Scrubber.tsx [hunk 1]
```diff
@@ -80,7 +79,10 @@ export default function Scrubber(props: Props): JSX.Element {
 
   const setHoverValue = useSetHoverValue();
 
-  const [hoverStamp, setHoverStamp] = useState<Time | undefined>();
+  const [hoverInfo, setHoverInfo] = useState<
+    { stamp: Time; position: { clientX: number; clientY: number } } | undefined
```
Comment (defunctzombie) [inline packages/studio-base/src/components/PlaybackControls/Scrubber.tsx:83]: Tempted to say that this type has evolved beyond what we should have inline and should hoist to a `type` in the file for readability.

File: packages/studio-base/src/components/PlaybackControls/Scrubber.tsx [hunk 2]
```diff
@@ -147,57 +150,60 @@ export default function Scrubber(props: Props): JSX.Element {
...
+  const isHovered = hoverInfo != undefined;
+  const popperProps: Partial<PopperProps> = useMemo(
+    () => ({
+      open: isHovered, // Keep the tooltip visible while dragging even when the mouse is outside the playback bar
+      popperRef,
+      modifiers: [
+        {
+          name: "computeStyles",
+          options: {
+            gpuAcceleration: false, // Fixes hairline seam on arrow in chrome.
```
Comment (defunctzombie) [inline packages/studio-base/src/components/PlaybackControls/Scrubber.tsx:162]: 😮

File: packages/studio-base/src/components/PlaybackControls/Slider.tsx [hunk 3]
```diff
@@ -12,7 +12,11 @@ type Props = {
   fraction: number | undefined;
   disabled?: boolean;
   onChange: (value: number) => void;
-  onHoverOver?: (value: number) => void;
+  /**
+   * @param value Hovered `fraction` value
+   * @param position Current hovered position in client coordinates
+   */
+  onHoverOver?: (value: number, position: { clientX: number; clientY: number }) => void;
```
Comment (defunctzombie) [inline packages/studio-base/src/components/PlaybackControls/Slider.tsx:19]: Once this starts to grow beyond one parameter maybe worth having a `HoverOverEvent`?

**Review by defunctzombie — APPROVED** [review]

File: packages/studio-base/src/components/PlaybackControls/Scrubber.tsx (hunk 2 above)
Comment (jtbandes) [reply packages/studio-base/src/components/PlaybackControls/Scrubber.tsx:162]: just moved

**Pull request merged**